server/src/test/java/org/opensearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -307,38 +332,63 @@ public void testSimple() throws Exception {
         }
     }
 
+    /**
+     * This test case utilizes the default implementation of GlobalOrdinalsStringTermsAggregator.
+     * In this case, the segment terms will not get initialized and will run without LeafBucketCollector#termDocFreqCollector optimization
+     */
     public void testStringIncludeExclude() throws Exception {
+        testStringIncludeExclude(
+            (document, field, value) -> document.add(new SortedSetDocValuesField(field, new BytesRef(value))),
+            DEFAULT_POST_COLLECTION
+        );
+    }
+
+    /**
+     * This test case utilizes the default implementation of GlobalOrdinalsStringTermsAggregator.
+     * In this case, the segment terms will get initialized and will use LeafBucketCollector#termDocFreqCollector optimization
+     */
+    public void testStringIncludeExcludeWithStoredValues() throws Exception {
+        // aggregator.postCollection() is not required when LeafBucketCollector#termDocFreqCollector optimization is used
+        // This also verifies that the bucket count is completed without running postCollection()
+        testStringIncludeExclude((document, field, value) -> {
+            document.add(new SortedSetDocValuesField(field, new BytesRef(value)));
+            document.add(new StringField(field, value, Field.Store.NO));
+        }, NOOP_POST_COLLECTION);
+    }
+
+    private void testStringIncludeExclude(TriConsumer<Document, String, String> addField, Consumer<Aggregator> postCollectionConsumer)
+        throws Exception {
         try (Directory directory = newDirectory()) {
             try (RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory)) {
                 Document document = new Document();
-                document.add(new SortedSetDocValuesField("mv_field", new BytesRef("val000")));
-                document.add(new SortedSetDocValuesField("mv_field", new BytesRef("val001")));
-                document.add(new SortedDocValuesField("sv_field", new BytesRef("val001")));
+                addField.apply(document, "mv_field", "val000");
+                addField.apply(document, "mv_field", "val001");
+                addField.apply(document, "sv_field", "val001");
```

Comment:
Previously, this test was mixing `SortedSetDocValues` and `SortedDocValues`. Now you're just adding one or the other. We're losing some coverage.

Reply:
I see, seems like I missed this when refactoring. Nice catch.